lib/chunked-passthrough.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { Transform } = require('stream');
+
+class ChunkedPassthrough extends Transform {
+    constructor(options = {}) {
+        let config = {
+            readableObjectMode: true,
+            writableObjectMode: false
+        };
+        super(config);
+        this.chunkSize = options.chunkSize || 64 * 1024; // 64KB default
+        this.buffer = Buffer.alloc(64 * 1024);
+    }
+
+    _transform(chunk, encoding, callback) {
+        this.buffer = Buffer.concat([this.buffer, chunk]);
+
+        while (this.buffer.length >= this.chunkSize) {
+            this.buffer = this.buffer.subarray(this.chunkSize);
```

Comment:
If it is larger than minimal then just send the entire buffer at once, no point in splitting it.